src/routes/transactions/handlers/preview.rs

```diff
@@ -0,0 +1,123 @@
+use rocket::futures::future::OptionFuture;
+use serde::{Deserialize, Serialize};
+
+use crate::common::models::addresses::AddressEx;
+use crate::common::models::backend::transactions::SafeTransaction;
+use crate::common::models::data_decoded::{DataDecoded, Operation};
+use crate::providers::info::{DefaultInfoProvider, InfoProvider};
+use crate::routes::contracts::models::DataDecoderRequest;
+use crate::routes::transactions::converters::details::is_trusted_delegate_call;
+use crate::routes::transactions::models::details::TransactionData;
+use crate::routes::transactions::models::TransactionInfo;
+use crate::utils::context::RequestContext;
+use crate::utils::errors::ApiResult;
+use crate::utils::http_client::Request;
+
+#[derive(Deserialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct TransactionPreviewRequest {
+    to: String,
+    data: Option<String>,
+    value: String,
+    operation: Operation,
+}
+
+#[derive(Serialize, Debug, PartialEq)]
+#[serde(rename_all = "camelCase")]
+#[cfg_attr(test, derive(Deserialize))]
+pub struct TransactionPreview {
+    pub tx_info: TransactionInfo,
+    pub tx_data: TransactionData,
+}
+
```

Comment:
The reason why I didn't do this is because I don't see the value right now on splitting this feature in more files.

I'd keep things as close as possible for a specific feature which allows us to better encapsulate them in the future (example to remove the `pub`).

But there's no strict rule regarding this. We can discuss how to proceed in a meeting 👍

Reply:
Ok, understood, it is a very specific model for this feature (preview) only then we could do an exception, migrate this to a new file in the future if was necessary is not too complicated.

Reply:
Exactly. Overall it's better to start with a smaller scope and as soon as we see that a new endpoint requires this structure we see where we should start creating a baseline for the "shared" models.

It'd be more difficult to start with the shared part and then see where we can have them private or not.